utils/general.py

```diff
@@ -952,9 +953,12 @@ def increment_dir(dir, comment=''):
     # Increments a directory runs/exp1 --> runs/exp2_comment
     n = 0  # number
     dir = str(Path(dir))  # os-agnostic
-    d = sorted(glob.glob(dir + '*'))  # directories
-    if len(d):
-        n = max([int(x[len(dir):x.rfind('_') if '_' in Path(x).name else None]) for x in d]) + 1  # increment
+    dirs = sorted(glob.glob(dir + '*'))  # directories
```

Comment:
I think a recent PR https://github.com/ultralytics/yolov5/pull/1058 may have addressed this issue. Can you verify that the problem still remains in master? What was the failure mode you mentioned before? I'll check your comments again.

Reply:
I think it does not solve it, moreover, the line is impossible to read...

Reply:
this will still fail for any folder which is not the experimental one like `my-sample_folder_here`

Reply:
Ok got it. Yes, I agree impossible to read.

I will try to update this PR then with the the #1058 updates as well.

Reply:
Ok seems like no changes are required due to updated code.